reana_server/validation.py

```diff
@@ -125,3 +134,20 @@ def validate_workflow(reana_yaml: Dict, input_parameters: Dict) -> None:
     validate_compute_backends(reana_yaml)
     validate_workspace_path(reana_yaml)
     validate_inputs(reana_yaml)
+
+
+def validate_retention_rule(rule: str) -> None:
+    """Validate retention rule.
+
+    :param rule: retention rule
+    :type rule: string
+
+    :raises reana_commons.errors.REANAValidationError: if rule is not valid
+    """
+    rule_path = pathlib.Path(rule)
```

Comment:
Do you think using `pathlib` can cause some problems given that we are validating patterns and not paths? In particular, I don't know exactly how the library manipulates paths and whether this can cause undesired behaviours when passing patterns with wildcards.